### PR TITLE
[WR-501] Adding role attributes

### PR DESCRIPTION
**Pull request by tozny-shakeeb**

### CODE REVIEW

File: tozny/resource_realm_role.go [hunk 1]
```diff
@@ -63,6 +63,31 @@ func resourceRealmRole() *schema.Resource {
...
+				Type:        schema.TypeList,
+				Optional:    true,
+				ForceNew:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"key": {
+							Description: "The key to use for the attribute",
+							Type:        schema.TypeString,
+							Required:    true,
+							ForceNew:    true,
```
Comment (rosasck) [inline tozny/resource_realm_role.go:77]: do u need to recreate or can u just update?

File: tozny/resource_realm_role.go [hunk 2]
```diff
@@ -63,6 +63,31 @@ func resourceRealmRole() *schema.Resource {
 				Computed:    true,
 				ForceNew:    true,
 			},
+			"attribute": {
+				Description: "Arbitrary string-string key value pairs.",
+				Type:        schema.TypeList,
+				Optional:    true,
+				ForceNew:    true,
+				Elem: &schema.Resource{
```
Comment (rosasck) [inline tozny/resource_realm_role.go:71]: if an attribute is given do we need to recreate? then we need to re add all users, can we not use update realm role?

File: tozny/resource_realm_role.go (hunk 1 above)
Comment (tozny-shakeeb) [reply tozny/resource_realm_role.go:77]: If the attributes exist then we would just need to update or else create them but these align with how the other attributes are defined as well so basically any change to role will eventually recreate the role

Comment (rosasck) [reply tozny/resource_realm_role.go:77]: if it matches how attributes are in groups and other files its okay

**Review by rosasck — APPROVED** [review]
they may ask us to do an update for this, since it would remove from all users

File: tozny/resource_realm_role.go (hunk 1 above)
Comment (tozny-shakeeb) [reply tozny/resource_realm_role.go:77]: yeah - I just used the same one from groups

**Review by rosasck — APPROVED** [review]
THAK YOUUUU

**Pull request merged**